lib/rpcs/v1_4_5/speculative_exec.rs

```diff
@@ -0,0 +1,35 @@
+use serde::{Deserialize, Serialize};
+
+use casper_types::ProtocolVersion;
+
+use crate::rpcs::common::BlockIdentifier;
+
+use crate::types::{Deploy, DeployHash};
+
+pub(crate) const SPECULATIVE_EXEC_METHOD: &str = "speculative_exec";
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub(crate) struct SpeculativeExecParams {
+    block_identifier: Option<BlockIdentifier>,
+    deploy: Deploy,
+}
+
+impl SpeculativeExecParams {
+    pub(crate) fn new(block_identifier: Option<BlockIdentifier>, deploy: Deploy) -> Self {
+        SpeculativeExecParams {
+            block_identifier,
+            deploy,
+        }
+    }
+}
+
+/// The `result` field of a successful JSON-RPC response to an `speculative_exec` request.
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct SpeculativeExecResult {
+    /// The JSON-RPC server version.
+    pub api_version: ProtocolVersion,
+    /// The deploy hash.
+    pub deploy_hash: DeployHash,
+}
```

Comment:
This needs to be the same as the type defined by the node [here](https://github.com/casper-network/casper-node/blob/cc95de3a80cdada1d67a84a2178f23bcf10e36ae/node/src/components/rpc_server/rpcs/speculative_exec.rs#L58-L66):
```suggestion
pub struct SpeculativeExecResult {
    /// The JSON-RPC server version.
    pub api_version: ProtocolVersion,
    /// Hash of the block on top of which the deploy was executed.
    pub block_hash: BlockHash,
    /// Result of the execution.
    pub execution_result: ExecutionResult,
}
```

Reply:
That would add a dependency on casper-node on account of BlockHash. What's the suggested workaround? IIRC, you don't want to introduce dependencies on node.